.circleci/config.yml

```diff
@@ -92,6 +92,44 @@ jobs:
             # retry introduced in https://github.com/algolia/react-instantsearch/pull/3475
             retry yarn test --maxWorkers=4
 
+  test_unit_react_v17:
+    <<: *defaults
+    steps:
+      - checkout
+      - run: *install_yarn_version
+      - restore_cache: *restore_yarn_cache
+      - run: *run_yarn_install
+      - save_cache: *save_yarn_cache
+      - run:
+          name: Install React v17 dependencies
+          command: |
+            yarn add --force -DW \
+              react@17.0.2 \
```

Comment:
this seems like a decent solution for now, but I wonder if we could have a separate package with those dependencies, and then run the tests:

```
└ tests-react-17
  └ package.json with react 17 dependencies
  └ jest ../jest.config.js
```

That way the node_modules used should be the one from react 17 (maybe also similar possible with algoliasearch v3), but tbh I'm not fully sure if that would work